### ISSUE TITLE
February Retrospective

### ISSUE THREAD

**Issue by aaleksandraristic** [IanTDuncan/MealTime#118]
February Retrospective Details
---------------------------------------------------

Description:
------------------------------

This is the issue for the February Retrospective where we can talk and summarize what things went well and what went wrong during this month's work.

- @BeepDroid @Cade5480 @bdshanks-28 @IanTDuncan 

Retrospective Summary:
------------------------------

**Went wrong:**

- [ ] How can we explain the problem that you @IanTDuncan  had with APIs?
- [ ] @bdshanks-28  did you have any problems? I think there was something during this month but I can't remember specifically.
------------------------------------------------
- [ ] I didn't organize my team to create all the issues (for every step/action that they're working on) and set up due dates on time.
- [ ] Furthermore, all of us started creating the issues without the plan so there has appeared a problem of redundant issues.
- [ ] Brandon assigned himself a too many issues to the same date so his work due dates had to be extended.
- [ ] Brandon also didn't create enough issues to track all the changes that he's making (actions/cleaning the issues and so on.)
- [ ] Cade's (Backend dev issues) and Brandon's (XML issues) issues were too broad and contained a lot of information.
- [ ] Updating due dates and GitHub issues didn't work that well at the beginning. Especially for Brandon's issues. I had also expended some of the issues for Cade. 
- [ ] Not following the format for the GitHub issues (Brandon, Cade, Ian)
- [ ] In-person meeting attendance - Usually Kerry, Ian, Cade and I; 
- [ ] Ian had problem with APIs - unfamiliar with it and didn't know how to approach them (mistakes with choosing what type of code to implement)

**Went well:**
- [ ] Team in-person meetings
- [ ] Good communication between team members
- [ ] Kerry's involvement and helping other team members
- [ ] Good teamwork, each team member was dedicated and responsible for their tasks
- [ ] Quickly addressing and fixing issues

**Goals for the upcoming month:**
To finalize the database part and the Firebase RealTime Database so we can start testing and analyzing our app. 
To have a working prototype by the first half/sooner in March, and then start debugging.
To get our MVP as soon as possible. 
Keep this working speed, plan and organization (which is much better and clear); complete all the issues and get our product (app) ready.

**Comment by bdshanks-28** [comment]
Went Wrong:

What went wrong for me was that when we started assigning due dates, I had too many issues assigned due on the same date meaning when I finally got around to those issues, they were late. Another issue I had was I was not creating enough issues to track all the changes. I also wasn't correctly using the issue formatting though that was more confusion on my part of what specifics meant for the format.
________________________________
From: aaleksandraristic ***@***.***>
Sent: Sunday, February 25, 2024 10:31:23 PM
To: IanTDuncan/MealTime ***@***.***>
Cc: shankleb16 ***@***.***>; Mention ***@***.***>
Subject: [IanTDuncan/MealTime] February Retrospective (Issue #118)


February Retrospective Details
Description:

This is the issue for the February Retrospective where we can talk and summarize what things went well and what went wrong during this month's work.

  *   @BeepDroid<https://github.com/BeepDroid> @Cade5480<https://github.com/Cade5480> @bdshanks-28<https://github.com/bdshanks-28> @IanTDuncan<https://github.com/IanTDuncan>

Retrospective Summary:

Went wrong:

  *    How can we explain the problem that you @IanTDuncan<https://github.com/IanTDuncan> had with APIs?
  *    @bdshanks-28<https://github.com/bdshanks-28> did you have any problems? I think there was something during this month but I can't remember specifically.
  *    Updating due dates and GitHub issues didn't work that well at the beginning, now it's much better
  *    I probably need to say about the formatted comments (and point out specific members who had problems with it)

Went well:

  *    Good conversations between team members
  *    Good teamwork, especially Kerry helping out with other issues

**Comment by BeepDroid** [comment]
I think another thing that went wrong was the amount of issues we had and our broadness. We weren't very specific in some issues and some information got lumped in with bigger and bigger issues that was a little muddy to scroll through. 

However, our responsiveness was a good point I think! We were really quick to address and fix issues when they were pointed out to us! Alex did a great job of keeping us on the ball.

**Comment by IanTDuncan** [comment]
Sorry this is late but some of the problems I encountered with APIs was that I was entirely unfamiliar with and didn't know how to approach them. Because of this I made mistakes with choosing what type of code to implement and sorta kept on going around in a circle coding-wise. 

I also think the broad issues were a problem and we didn't really understand how to do them until recently. I believe if we started with specific issues, we would of begun developing the app much more smoothly. 

Overall, while we definitely got hurt in the long-run from our rocky start and big, general issues, I believe this month we showed actual improvement in terms of organization and are on-track.
